test/src/main/java/org/apache/accumulo/test/functional/BulkNewMetadataSkipIT.java

```diff
@@ -0,0 +1,224 @@
+ * tablets. This test also splits the metadata table such that the tablet metadata for each tablet
+ * of the test table is in its own metadata tablet. The test then runs with different values for the
+ * TABLE_BULK_SKIP_THRESHOLD property starting with zero (disabled) then increasing.
+ *
+ * This test uses AccumuloClusterHarness instead of SharedMiniClusterBase so that we don't have to
+ * re-merge the metadata table and delete the test table. Doing these two things, and then waiting
+ * for balancing, takes a long time. It's faster to just start with a clean instance for each test
+ * run.
+ */
+public class BulkNewMetadataSkipIT extends AccumuloClusterHarness {
+
+  @Override
+  public void configureMiniCluster(MiniAccumuloConfigImpl cfg, Configuration hadoopCoreSite) {
+    cfg.setMemory(ServerType.TABLET_SERVER, 512, MemoryUnit.MEGABYTE);
+    cfg.setProperty(Property.MANAGER_TABLET_GROUP_WATCHER_INTERVAL, "3s");
+    cfg.setProperty(Property.TSERV_ASSIGNMENT_MAXCONCURRENT, "25");
+    cfg.setNumTservers(1);
+    hadoopCoreSite.set("fs.file.impl", RawLocalFileSystem.class.getName());
+  }
+
+  public static String writeNonContiguousData(FileSystem fs, String file,
+      AccumuloConfiguration aconf, int[] rows) throws Exception {
+    String filename = file + RFile.EXTENSION;
+    try (FileSKVWriter writer = FileOperations.getInstance().newWriterBuilder()
+        .forFile(filename, fs, fs.getConf(), NoCryptoServiceFactory.NONE)
+        .withTableConfiguration(aconf).build()) {
+      writer.startDefaultLocalityGroup();
+      for (int i : rows) {
+        writer.append(new Key(new Text(row(i))), new Value(Integer.toString(i)));
+      }
+    }
+    return hash(filename);
+  }
+
+  @BeforeEach
+  @Override
+  public void setupCluster() throws Exception {
+    super.setupCluster();
+    // prime the zk connection
+    try (AccumuloClient c = Accumulo.newClient().from(getClientProps()).build()) {}
+  }
+
+  @ParameterizedTest
+  @ValueSource(ints = {0, 0, 2, 4, 8, 16, 32, 64, 128})
+  public void test(int skipDistance) throws Exception {
+
+    final String tableName = getUniqueNames(1)[0] + "_" + skipDistance;
+    final AccumuloConfiguration aconf = getCluster().getServerContext().getConfiguration();
+    final FileSystem fs = getCluster().getFileSystem();
+    final String rootPath = getCluster().getTemporaryPath().toString();
+    final String dir = rootPath + "/" + tableName;
+
+    fs.delete(new Path(dir), true);
+
+    final SortedSet<Text> splits = new TreeSet<>();
+    IntStream.rangeClosed(0, 1000).forEach(i -> splits.add(new Text(String.format("%04d", i))));
+
+    final NewTableConfiguration ntc = new NewTableConfiguration();
+    ntc.setProperties(
+        Map.of(Property.TABLE_BULK_SKIP_THRESHOLD.getKey(), Integer.toString(skipDistance)));
+    ntc.withSplits(splits);
+
+    final Map<String,Set<String>> hashes = new HashMap<>();
+    IntStream.rangeClosed(0, 1000).forEach(i -> hashes.put(row(i) + "", new HashSet<>()));
```

Comment:
```suggestion
    IntStream.rangeClosed(0, 1000).forEach(i -> hashes.put(row(i), new HashSet<>()));
```
Is the empty string concatenation needed since row always returns a formatted string?
```
public static String row(int r) {
    return String.format("%04d", r);
}
```

Reply:
I think that's leftover from old code. I'll apply your fix.